File: 0260_Problem_1.cpp
// Problem Link: https://leetcode.com/contest/weekly-contest-260/problems/maximum-difference-between-increasing-elements/
// Godbolt Link: https://godbolt.org/z/3ze4WhEMG

auto maximum_difference(std::vector<int> nums) -> int {
    int result     = -1;
    int min_so_far = std::numeric_limits<int>::max();
    for (auto elem : nums) {
        min_so_far = std::min(min_so_far, elem);
        if (elem > min_so_far) {
            result = std::max(result, elem - min_so_far);
        }
    }
    return result;
}

File: 0260_Problem_1.py
# Problem Link: https://leetcode.com/contest/weekly-contest-260/problems/maximum-difference-between-increasing-elements/

import sys

def maximum_difference(nums):
    result, min_so_far = -1, sys.maxsize
    for elem in nums:
        min_so_far = min(min_so_far, elem)
        if elem > min_so_far:
            result = max(result, elem - min_so_far)
    return result

# More concise variable names
def maximum_difference(nums):
    res, lo = -1, sys.maxsize
    for e in nums:
        lo = min(lo, e)
        if e > lo:
            res = max(res, e - lo)
    return res
